src/main/java/com/internet/shop/dao/util/HashUtil.java

```diff
@@ -0,0 +1,31 @@
+package com.internet.shop.dao.util;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
+    private static final String HASH_ALGORITHM = "SHA-512";
+
+    public static String hashPassword(String password, byte[] salt) {
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(HASH_ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new IllegalArgumentException("Couldn't hash password", e);
```

Comment:
```suggestion
            throw new RuntimeException("Couldn't hash password", e);
```
let's throw a more generic Exception - I think IllegalArgumentException does not suit here well